lib/checks/navigation/identical-links-same-purpose-after.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Skip unless there are more than a single result
+ */
+if (results.length < 2) {
+	return results;
+}
+
+/**
+ * for each result
+ * - get other results with matching accessible name
+ * - check if same purpose is served
+ *  - if not change `result` to `undefined`
+ */
+for (let index = 0; index < results.length; index++) {
+	const { data: nodeData } = results[index];
+	const { name, resource } = nodeData;
+	const identicalNodes = getNodesWithIdenticalAccessibleName(name, index);
+	if (!identicalNodes.length) {
+		continue;
+	}
+	if (!resource) {
+		results[index].result = undefined;
+		continue;
+	}
+	if (!servesSamePurpose(resource, identicalNodes)) {
+		results[index].result = undefined;
+		continue;
+	}
+}
+
+return results;
+
+/**
+ * Get list of nodes from results which match a given accessible name
+ * @method getNodesWithIdenticalAccessibleName
+ * @param {String} expectedName accessible name to be matched
+ * @param {Number} excludeIndex exclude `index` of result, that should not be taken into consideration
+ * @returns {Array<Object>}
+ */
+function getNodesWithIdenticalAccessibleName(expectedName, excludeIndex) {
+	return results.filter((result, index) => {
```

Comment:
This would mean we do not raise `incomplete` on all the nodes that have identical accessible names. So we let one `pass` and raise `incomplete` on the other nodes. This needs to be discussed.

Reply:
`getNodesWithIdenticalAccessibleName` returns all nodes that are identical, we can mark all of them as incomplete in the same go instead of just the one you started on.